src/Bicep.Core.Samples/Files/InvalidModules_LF/main.syntax.bicep

```diff
@@ -2167,16 +2161,13 @@ module expectedComma 'modulea.bicep' = [for (x)]
 //@[039:00048) | └─ForSyntax
 //@[039:00040) |   ├─Token(LeftSquare) |[|
 //@[040:00043) |   ├─Token(Identifier) |for|
-//@[044:00047) |   ├─ForVariableBlockSyntax
-//@[044:00045) |   | ├─Token(LeftParen) |(|
-//@[045:00046) |   | ├─LocalVariableSyntax
-//@[045:00046) |   | | └─IdentifierSyntax
-//@[045:00046) |   | |   └─Token(Identifier) |x|
-//@[046:00046) |   | ├─SkippedTriviaSyntax
-//@[046:00046) |   | ├─LocalVariableSyntax
-//@[046:00046) |   | | └─IdentifierSyntax
-//@[046:00046) |   | |   └─SkippedTriviaSyntax
-//@[046:00047) |   | └─Token(RightParen) |)|
+//@[044:00047) |   ├─SkippedTriviaSyntax
+//@[044:00047) |   | └─VariableBlockSyntax
```

Comment:
Is the variable block a child node of `SkippedTriviaSyntax` because it is invalid?

Reply:
Yup, exactly. The current code has a `SkippedTriviaSyntax` inside the `ForVariableBlockSyntax` (one for "missing comma", and one for "missing 2nd identifier") to indicate that it's invalid (because the for syntax only supports **exactly** 2 arguments inside a set of parantheses:
![image](https://user-images.githubusercontent.com/38542602/183134319-78aa0aa9-f5dc-421c-9a1b-d074a139ef07.png)

The new approach parses the block and says "this is a valid parenthesized argument block", but then wraps it in skipped trivia to indicate that it is not valid, and to attach an error to it.